modules/oe_theme_content_event/src/Plugin/ExtraField/Display/OnlineDescriptionExtraField.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_content_event\Plugin\ExtraField\Display;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\oe_content_event\EventNodeWrapper;
+
+/**
+ * Extra field displaying online description on events.
+ *
+ * @ExtraFieldDisplay(
+ *   id = "oe_theme_content_event_online_description",
+ *   label = @Translation("Online description"),
+ *   bundles = {
+ *     "node.oe_event",
+ *   },
+ *   visible = true
+ * )
+ */
+class OnlineDescriptionExtraField extends DateAwareExtraFieldBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(ContentEntityInterface $entity) {
+    $event = EventNodeWrapper::getInstance($entity);
+
+    if (!$event->hasOnlineType() || !$event->hasOnlineLink() || !$event->hasOnlineDates()) {
+      // All online fields have to be filled to show online information.
+      return [];
+    }
+
+    $build = [];
+
+    if ($event->isOnlinePeriodYetToCome($this->requestDateTime)) {
+      // Invalidate cache tags when livestream will be started.
+      $this->applyHourTag($build, $event->getOnlineStartDate());
```

Comment:
This is correct and it's basically what outlined in the comment above, on the other field, but we should consider marking the field as empty, instead of returning and empty array, as per extra field docs:

![image](https://user-images.githubusercontent.com/153362/146939311-b27af6a3-285b-4c38-94eb-85793ad8d239.png)